### ISSUE TITLE
"over-weighted" prompts causing damage to 2.1 outputs?

### ISSUE THREAD

**Issue by bghira** [damian0815/compel#35]
Hello! I have a bot that can generate prompts and I've done few-shot learning to teach it how to generate Compel prompts, but quite honestly it is very bad, and sometimes it overly weights or simultaneously deemphasizes the same token, example prompt:

```
Create a prompt: absurdist-metaphorical++-objects turned.--
```

resulting images:

![image](https://github.com/damian0815/compel/assets/59658056/e7177d58-8462-4f16-817a-08b736dc40be)
![image](https://github.com/damian0815/compel/assets/59658056/2a501083-7b5a-4fa6-aa1c-87708c941ef9)

It's resolution/seed dependent, as well. This can be avoided by changing the prompt, or simply not having the bot do the emphasis at all. But I wanted to bring it up anyway, just in case it's incorrect behaviour.

Here's another example:

```
An astronaut--sitting in a teacup++ floating through space++
```

![image](https://github.com/damian0815/compel/assets/59658056/167787c0-d099-4b88-bfaf-b4cd5b492779)
![image](https://github.com/damian0815/compel/assets/59658056/52719da6-ef26-445f-a994-ea2362b9b16b)

Settings:

DDPMScheduler, 32 steps, cfg 7.2, seed 789420774 (probably cursed), resolution 1024x1536 (though it happens at every resolution)

some examples of functional outputs that verify the model actually works:
![image](https://github.com/damian0815/compel/assets/59658056/c3588d32-e18c-4dbf-b67a-6342e679edf0) ![image](https://github.com/damian0815/compel/assets/59658056/de3d213b-4645-4528-aebb-f0c5d42cb535) ![image](https://github.com/damian0815/compel/assets/59658056/22e18236-fd90-4637-b66a-650ff2f9da7a)

**Comment by damian0815** [comment]
SD2.1 actually seems to respond poorly to weighting, both positive and negative. i'm working on something that will allow multiple prompts which is a game changer for sd 2.1 - instead of having a single prompt you can have multiple prompts that get joined using torch.cat(). syntax will be something like `("a bird sitting on a branch", "a painting by van gogh").concat(1,1)`

**Comment by damian0815** [comment]
in the meantime try your prompts without the weights and see what you get, or try manually increasing the weights from 1 to 1.2 to see if there's a phase change somewhere, ie `a (metaphorical)1.01 teapot` through to `a (metaphorical)1.3 teapot` or whatever

**Comment by bghira** [comment]
to clarify, would it be useful for that testing on your side? or would it not help much?

**Comment by damian0815** [comment]
thanks but no, i see this as an issue with SD more generally, not with parsing weights.

**Comment by bghira** [comment]
i've re-trained from 2.1-v again more carefully this time, and far slower. for context, that checkpoint i opened this issue about was trained to 4200 steps in about a day. the new checkpoint hit 4200 steps in 4 days.

the batch size matters a LOT when fine-tuning 2.1. that, and the number of layers you freeze. all but the last two layers being frozen seems to prevent this problem.

**Comment by damian0815** [comment]
> i've re-trained from 2.1-v again more carefully this time, and far slower. for context, that checkpoint i opened this issue about was trained to 4200 steps in about a day. the new checkpoint hit 4200 steps in 4 days.
> 
> the batch size matters a LOT when fine-tuning 2.1. that, and the number of layers you freeze. all but the last two layers being frozen seems to prevent this problem.

are you on the EveryDream2 discord? i came to the same conclusion re: layer freezing (down the number of layers to freeze!). what batch size are you seeing? i'm seeing good results with 24 (batch size 6 + grad accum 4), curious if you're going higher..
